src/link/Elf.zig

```diff
@@ -3261,3 +3227,145 @@ fn padToIdeal(actual_size: anytype) @TypeOf(actual_size) {
     return std.math.add(@TypeOf(actual_size), actual_size, actual_size / ideal_factor) catch
         std.math.maxInt(@TypeOf(actual_size));
 }
+
+// Compute libc runtime initialization objects for a given link command.
+const CrtObjects = struct {
+    crt0: ?[]const u8 = null,
+    crti: ?[]const u8 = null,
+    crtbegin: ?[]const u8 = null,
+    crtend: ?[]const u8 = null,
+    crtn: ?[]const u8 = null,
+
+    fn compute(self: *CrtObjects, link_options: link.Options, comp: *const Compilation) void {
```

Comment:
Nit, what about `fn compute(self: *CrtObjects, link_options: link.Options, comp: *const Compilation) CrtObjects` ?
This way you're explicitly stating that `CrtObjects` by itself is uninitialized.